packages/sdk/src/wallet/providers/privy.ts

```diff
@@ -55,8 +56,8 @@ export class PrivyWalletProvider implements WalletProvider {
       // TODO: Implement proper user-to-wallet lookup
       const wallet = await this.privy.walletApi.getWallet({ id: userId })
 
-      const walletInstance = new Wallet(wallet.id)
-      walletInstance.address = wallet.address as Address
+      const walletInstance = new Wallet(wallet.id, this.tokenRegistry)
+      walletInstance.init(getAddress(wallet.address))
```

Comment:
I could see us eventually rewriting the wallet creation code, it's a bit intuitive.